src/test/java/org/jenkinsci/plugins/badge/WorkflowBadgeActionTest.java

```diff
@@ -0,0 +1,62 @@
+package org.jenkinsci.plugins.badge;
+
+
+import hudson.security.GlobalMatrixAuthorizationStrategy;
+import hudson.security.SecurityRealm;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.PresetData;
+
+
+/**
+ * Just the fun bits: check that actions register correctly
+ */
+public class WorkflowBadgeActionTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @PresetData(PresetData.DataSet.NO_ANONYMOUS_READACCESS)
+    @Test
+    public void authenticatedAccess() throws Exception {
```

Comment:
Most of this just seems to duplicate checks which are already being done in `PublicBadgeActionTest`. IMO it would suffice to have a single test case which verifies that you can get status on a `WorkflowJob` or `WorkflowRun` generally, without paying attention to security.